WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -194,28 +196,26 @@ class StatsInsightsStore: QueryStore<InsightStoreState, InsightQuery> {
         processQueries()
     }
 
-    func persistToCoreData() {
-        guard
-            let siteID = SiteStatsInformation.sharedInstance.siteID,
-            let blog = Blog.lookup(withID: siteID, in: ContextManager.shared.mainContext) else {
-                return
+    func saveDataInCache() {
+        guard let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
         }
```

Comment:
Since we only need to store and read cache for one site, do you think we can add a function to the cache type which removes all the cache that does not belong to the `siteID`? So that we can call the new "purge cache" function here, instead of when log out in `AccountService`. I see two main benefits in doing that:
1. The stats cache related stuff remains at the Stats part of the app. It'd be easier to read in a few months time (we may forgot cache is cleaned up after logout).
2. Users may switch sites without logging out. We can make the cache more efficient by only store current site's cache.

Reply:
Is there a good place to put this logic? I found `onBlogSwitched` but most of the logic is in the dashboard view controller. I'm not sure it's a good place to put it. 

I'd rather just keep it in memory in case the user switches back to the previous site. The stats mostly just contain numbers – they shouldn't take up a lot of space in memory.

The cache is cleared on logout to address the following comment from the [previous PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/21059#discussion_r1260495975):

> I'm not sure if the app would delete all the files in the cache directory upon logout. If it doesn't, we may has an issue where cache saved in one account is used by another account.

Reply:
Clearing cache upon logout is okay. We do many other cleaning up things during log out. I just thought "self cleaning up" might be a better alternative. That's why suggested purging cache within the Stats part of the app, instead of relying other code.

One way to implement it would be: adding a `StatsInsightsCache.removeCache(keepingSite: SiteID)` and call it at this `saveDataInCache` function here, so that there is only one site's cache is kept in any given time. Same can be done to `StatsPediodCache` too.